stockflux-securities-master/src/components/Alert.js

```diff
@@ -2,31 +2,39 @@ import React from "react";
 import "./Alert.css";
 import PropTypes from "prop-types";
 
-const Alert = ({ messages, type, className }) => {
-  return (
-    <div className={`securities-alert-container ${className}`}>
-      {Array.isArray(messages) && messages.length > 1 ? (
-        <ul>
-          {messages.map((message, index) => (
-            <li className={`securities-alert ${type}`} key={index}>
-              {message}
-            </li>
-          ))}
-        </ul>
-      ) : (
-        <div className={`securities-alert ${type}`}>{messages}</div>
-      )}
-    </div>
-  );
+export function AlertType(value) {
+  this._value = value;
+}
+
+AlertType.prototype.valueOf = function() {
+  return this._value;
 };
 
+AlertType.ERROR = new AlertType("error");
+AlertType.SUCCESS = new AlertType("success");
+
+const Alert = ({ messages, type }) => (
+  <div className="securities-alert-container">
+    {Array.isArray(messages) && messages.length > 1 ? (
+      <ul>
+        {messages.map((message, index) => (
+          <li className={`securities-alert ${type.valueOf()}`} key={index}>
```

Comment:
I think the base element of the alert should be the one to house the base CSS classes. In this case, the `ul` should be the element that gets this `className` and your CSS should be changed to reflect. If you want to style elements that fall within the base component, then you can use CSS to target elements within them either by class, or by element type, e.g.

    .securities-alert {
      font-size: var(--font-xlarge);
    }
    .securities-alert li {
      margin: var(--margin-small);
    }
    .securities-alert.error {
      color: var(--error-red);
    }

Reply:
Okay, if you think it'a better that way. Because the styles are being applied to the list items or the single div I'll need to add a separate className for each :
`.securities-alert-list > li,
.securities-alert {
  font-size: var(--font-xlarge);
  margin: var(--margin-small);
}

.securities-alert-list.error > li,
.securities-alert.error {
  color: var(--error-red);
}

.securities-alert-list.success > li,
.securities-alert.success {
  color: var(--success-green);
}
`
`{Array.isArray(messages) && messages.length > 1 ? (
      <ul className={`securities-alert-list ${type.valueOf()}`}>
        {messages.map((message, index) => (
          <li key={index}>{message}</li>
        ))}
      </ul>
    ) : (
      <div className={`securities-alert ${type.valueOf()}`}>{messages}</div>
    )}`